IntoOOP/Coder/AlphabetCoder.cs

```diff
@@ -0,0 +1,103 @@
+﻿namespace IntoOOP.Coder;
+
+/// <summary>Класс, описывающий алфавитный шифратор строк.</summary>
+public abstract class AlphabetCoder : ICoder
+{
+    /// <summary>Список включенных алфавитов.</summary>
+    private List<Alphabet> _Alphabets = new();
+
+    /// <summary>Добавление алфавита.</summary>
+    /// <param name="alphabet">Алфавит для добавления.</param>
+    /// <returns>False, если алфавит не добавлен из-за пересечения 
+    /// символов с уже включенным алфавитом.</returns>
+    public bool AddAlphabet(Alphabet alphabet)
+    {
+        if (_Alphabets.Count == 0)
+        { 
+            _Alphabets.Add(alphabet);
+            return true;
+        }
+
+        if (Intersection(alphabet) != null) return false;
+
+        _Alphabets.Add(alphabet);
+        return true;
+    }
+
+    /// <summary>Добавление алфавитов.</summary>
+    /// <param name="alphabets">Алфавиты.</param>
+    /// <returns>False, если хотя бы один алфавит не добавлен из-за 
+    /// пересечения символов с уже включенным алфавитом.</returns>
+    public bool AddAlphabets(params Alphabet[] alphabets)
+    {
+        var result = true;
+
+        foreach (var alphabet in alphabets)
+            if (!AddAlphabet(alphabet)) result = false;
+            
+        return result;
```

Comment:
```suggestion
        foreach (var alphabet in alphabets)
            if (!AddAlphabet(alphabet)) 
                return false; // как только, так сразу!
            
        return true;
```
Как только нашли первое выполнение условия, сразу возвращаемся из метода. Зачем продолжать цикл?